packages/aws-cdk/lib/logging.ts

```diff
@@ -108,105 +46,239 @@ export async function withCorkedLogging<T>(block: () => Promise<T>): Promise<T>
   } finally {
     CORK_COUNTER--;
     if (CORK_COUNTER === 0) {
-      logBuffer.forEach(([stream, str]) => stream.write(str + '\n'));
+      // Process each buffered message through notify
+      for (const ioMessage of logBuffer) {
```

Comment:
Backlog item for Corking. Because Corking is global and we cannot have that.

Reply:
https://github.com/aws/aws-cdk/issues/32898